src/data-fetchers/bed/bed-parser.ts

```diff
@@ -0,0 +1,136 @@
+        fieldsToConvert.forEach(field => {
+            if (bedRecord[field]) bedRecord[field] = relativeToCumulative(bedRecord[field] as number, chromStart);
+        });
+        return bedRecord;
+    }
+    /**
+     * Generates an autoSql schema for a BED file that has custom columns
+     * @returns A string which is the autoSql spec
+     */
+    constructBedAutoSql() {
+        const AUTO_SQL_HEADER = `table customBedSchema\n"BED12"\n    (\n`;
+        const AUTO_SQL_FOOTER = '\n    )';
+
+        const autoSqlFields = this.#generateAutoSQLFields();
+        return String.prototype.concat(AUTO_SQL_HEADER, autoSqlFields, AUTO_SQL_FOOTER);
+    }
+    /**
+     * Generates the fields used in the autoSql schema. For custom column names.
+     * @returns A string which is are the fields in the autoSql schema
+     */
+    #generateAutoSQLFields() {
+        const BED12Fields: FieldInfo[] = [
+            ['string', 'chrom'],
+            ['uint', 'chromStart'],
+            ['uint', 'chromEnd'],
+            ['string', 'name'],
+            ['float', 'score'],
+            ['char', 'strand'],
+            ['uint', 'thickStart'],
+            ['uint', 'thickEnd'],
+            ['string', 'itemRgb'],
+            ['uint', 'blockCount'],
+            ['uint[blockCount]', 'blockSizes'],
+            ['uint[blockCount]', 'blockStarts']
+        ];
+        if (!this.#n_columns) throw new Error('Number of columns was not able to be determined');
+        if (!this.#customFields) return ''; // This function should never be called if there are no custom fields
+        const customFieldType = 'string';
+        const customFieldsWithTypes = this.#customFields.map(column => [customFieldType, column] as FieldInfo);
+
+        let allFields: FieldInfo[];
+        const REQUIRED_COLS = 3;
+        if (this.#n_columns > BED12Fields.length) {
+            if (this.#n_columns !== BED12Fields.length + this.#customFields.length) {
+                throw new Error(`BED file error: unexpected number of custom fields. Found ${this.#n_columns} columns 
+                    which is different from the expected ${BED12Fields.length + this.#customFields.length}`);
+            }
+            allFields = BED12Fields.concat(customFieldsWithTypes);
+        } else {
+            if (this.#n_columns - this.#customFields.length >= REQUIRED_COLS) {
+                allFields = BED12Fields.slice(0, this.#n_columns - this.#customFields.length).concat(
+                    customFieldsWithTypes
+                );
+            } else {
+                throw new Error(
+                    `Expected ${
+                        REQUIRED_COLS + this.#customFields.length
+                    } columns (${REQUIRED_COLS} required columns and ${
+                        this.#customFields.length
+                    } custom columns) but found ${this.#n_columns} columns`
+                );
+            }
+        }
```

Comment:
How about simpler logic?

```suggestion
        if(this.#n_columns - this.#customFields.length < REQUIRED_COLS) {
            throw new Error(
                `Expected ${
                    REQUIRED_COLS + this.#customFields.length
                } columns (${REQUIRED_COLS} required columns and ${
                    this.#customFields.length
                } custom columns) but found ${this.#n_columns} columns`
            );
        } else if(this.#n_columns !== BED12Fields.length + this.#customFields.length) {
            throw new Error(`BED file error: unexpected number of custom fields. Found ${this.#n_columns} columns 
                    which is different from the expected ${BED12Fields.length + this.#customFields.length}`);
        } else {
            allFields = BED12Fields.slice(0, this.#n_columns - this.#customFields.length).concat(
                customFieldsWithTypes
            );
        }
```

Reply:
Much cleaner! Thank you.